File: tests/fixtures/sample.ts
interface User {
  id: number;
  name: string;
  email: string;
}

export class AuthService {
  private token: string | null = null;

  async login(username: string, password: string): Promise<boolean> {
    // Authenticate with the API
    const response = await fetch('/api/auth', {
      method: 'POST',
      body: JSON.stringify({ username, password }),
    });
    if (response.ok) {
      this.token = await response.text();
      return true;
    }
    return false;
  }

  getToken(): string | null {
    return this.token;
  }
}

export async function fetchUser(id: number): Promise<User> {
  const response = await fetch(`/api/users/${id}`);
  return response.json();
}

export function validateEmail(email: string): boolean {
  const re = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return re.test(email);
}

File: tests/fixtures/sample.py
"""Sample Python module for testing."""

import os
from pathlib import Path


def authenticate_user(username: str, password: str) -> bool:
    """Authenticate a user with username and password."""
    if not username or not password:
        return False
    return username == "admin" and password == "secret"


class UserService:
    """Service for managing users."""

    def __init__(self, db_connection):
        self.db = db_connection

    def get_user(self, user_id: int) -> dict:
        """Fetch a user by ID."""
        return self.db.query(f"SELECT * FROM users WHERE id = {user_id}")

    def create_user(self, name: str, email: str) -> dict:
        """Create a new user."""
        return {"name": name, "email": email}


def handle_error(error: Exception) -> str:
    """Handle an error and return a message."""
    return f"Error occurred: {error}"
